Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -0,0 +1,902 @@
+            Real const *const AMREX_RESTRICT coefs_y = m_stencil_coefs_y.dataPtr();
+            int const n_coefs_y = m_stencil_coefs_y.size();
+            Real const *const AMREX_RESTRICT coefs_z = m_stencil_coefs_z.dataPtr();
+            int const n_coefs_z = m_stencil_coefs_z.size();
+
+            // extract tileboxes for which to loop
+            Box const &tbx = mfi.tilebox(Hfield[0]->ixType().toIntVect()); /* just define which grid type */
+            Box const &tby = mfi.tilebox(Hfield[1]->ixType().toIntVect());
+            Box const &tbz = mfi.tilebox(Hfield[2]->ixType().toIntVect());
+
+            // loop over cells and update fields
+            amrex::ParallelFor(tbx, tby, tbz,
+                [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+
+                    Real mag_Ms_arrx    = MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr);
+                    Real mag_alpha_arrx = MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_alpha_arr);
+                    Real mag_gamma_arrx = MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_gamma_arr);
+                
+                    // determine if the material is nonmagnetic or not
+                    if (mag_Ms_arrx != 0 && mag_alpha_arrx != 0 && mag_gamma_arrx != 0){
+                        // when working on M_xface(i,j,k, 0:2) we have direct access to M_xface(i,j,k,0:2) and Hx(i,j,k)
+                        // Hy and Hz can be acquired by interpolation
+
+                        // H_bias
+                        Real Hx_eff = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(1, 0, 0), amrex::IntVect(1, 0, 0), Hx_bias);
+                        Real Hy_eff = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0, 1, 0), amrex::IntVect(1, 0, 0), Hy_bias);
+                        Real Hz_eff = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0, 0, 1), amrex::IntVect(1, 0, 0), Hz_bias);
+
+                        if (coupling == 1){
+                            // H_eff = H_maxwell + H_bias + H_exchange + H_anisotropy ... (only the first two terms are considered here)
+
+                            // H_maxwell
+                            Hx_eff += MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(1, 0, 0), amrex::IntVect(1, 0, 0), Hx);
+                            Hy_eff += MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0, 1, 0), amrex::IntVect(1, 0, 0), Hy);
+                            Hz_eff += MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0, 0, 1), amrex::IntVect(1, 0, 0), Hz);
+                        }
+
+                        // calculate the a_temp_static_coeff (it is divided by 2.0 because the input dt is actually dt/2.0)
+                        Real a_temp_dynamic_coeff = PhysConst::mu0 * amrex::Math::abs(mag_gamma_arrx) / 2.0;
```

Comment:
Certain versions of our notes contain
-gamma
and others contain
|gamma|
Note gamma is always negative and we check (and abort) if this is not the case.
So either abs(mag_gamma) or -mag_gamma are both correct; I defer to Jackie here and we should make sure the paper has consistent notation.

Reply:
Nice catch! To be consistent, I think we would like to keep using `|gamma|`, instead of mix-using `|gamma|` and 
`-gamma`. Can we make this an issue to fix in the next PR? I'm afraid if I change it now I may introduce errors which is unnecessary especially since this PR is at such a late stage.

Reply:
I don't think it needs to be changed or addressed in a future PR.  I'll just update the overleaf.

Reply:
I want to point out in the paper equations (26) and (27), which come from (24), look funny because (26) uses |gamma| and (27) uses gamma.  Note that the code and paper are using the same conventions though, so it's technically fine as is.

Reply:
Yes, that's why we would like to be consistent. As least in the paper we want to keep using one notation, instead of both

Reply:
this is now consistently abs(gamma) everywhere -- consistent with the paper